integration/test/performancetest/transmitter.go

```diff
@@ -153,12 +197,182 @@ func (transmitter *TransmitterAPI) TableExist() (bool, error) {
 /*
 SendItem()
 Desc: Parses the input data and adds it to the dynamo table
-Param: data []byte is the data collected by data collector
+Param: packet map[string]interface{} is the data collected by data collector
 */
 func (transmitter *TransmitterAPI) SendItem(packet map[string]interface{}) (string, error) {
-	//this is currently a passthrough function and will change to have functionality with Okan's PR
-	//@TODO
-	sentItem, err := transmitter.AddItem(packet)
+	var sentItem string
+	var ae *types.ConditionalCheckFailedException // this exception represent the atomic check has failed
+	// check if hash exists
+	currentItem,err := transmitter.Query(packet[HASH].(string))
+	if err!=nil {
+		return "",err
+	}
+	if len(currentItem)==0{ // if an item with the same hash doesn't exist add it
+		sentItem, err = transmitter.AddItem(packet) 
+		// this may be overwritten by other test threads, in that case it will return a specific error
+	
+		if !errors.As(err,&ae){ // check if our add call got overwritten by other threads
+			return sentItem,err
+		}
+		if err!=nil { //any other error dont try again
+			return "",err
+		}
+		// addItem failed due to a competing thread
+		// instead of adding, proceed to update the item, with the same data
+		rand.Seed(time.Now().UnixNano())
+		time.Sleep(time.Duration(rand.Intn(UPDATE_DELAY_THRESHOLD))*time.Second)
+		fmt.Println("Item already exist going to update",len(currentItem))	
+	}
+	// item already exist so update the item instead
+	err = transmitter.UpdateItem(packet) //try to update the item
+	//this may be overwritten by other test threads, in that case it will return a specific error
+	if err !=nil{
+		return "",err
+	}
+	fmt.Println("SendItem Completed")
 	return sentItem, err
+
 }
+/*
+PacketMerger()
+Desc:
+	This function updates the currentPacket with the unique parts of newPacket and returns in dynamo format
+Params:
+	newPacket: this is the agentData collected in this test
+	currentPacket: this is the agentData stored in dynamo currently
+*/
+func (transmitter * TransmitterAPI) PacketMerger(newPacket map[string]interface{}, currentPacket map[string]interface{}) (map[string]types.AttributeValue,error){
+	testSettings := fmt.Sprintf("%s-%s",os.Getenv(PERFORMANCE_NUMBER_OF_LOGS),os.Getenv(TPS))
```

Comment:
This is the only place that uses `os.Getenv(TPS)`. This string should just be passed in as a parameter. We don't have any reason to persist the TPS or number of log files as an env var.

Reply:
I wanted to have it that so that it will be easier other configs, by just adding another os.Getenv into that sprintf